### ISSUE TITLE
BUG:Timezone lost when assigning Datetime via DataFrame.at

### ISSUE THREAD

**Issue by torfsen** [pandas-dev/pandas#33544]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import datetime as dt
import pandas as pd

df = pd.DataFrame({'foo': [dt.datetime(2000, 1, 1)]})
df.at[0, 'foo'] = dt.datetime(2000, 1, 2, tzinfo=dt.timezone.utc)
print(df.at[0, 'foo'])  # Prints 2000-01-02 00:00:00
assert df.at[0, 'foo'].tzinfo is not None  # Fails
```

#### Problem description

When assigning a tz-aware datetime via `at` I expect that the tz-information is kept. If that is not possible due to a datatype problem then I expect a warning or error message. Instead it is lost without a warning or error message.

#### Expected Output

I would have expected that the tz-information is kept, i.e. that

```
assert df.at[0, 'foo'].tzinfo is not None  # Should not fail
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.3.0-46-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.3.post20200330
Cython           : None
pytest           : 5.4.1
hypothesis       : None
sphinx           : 2.4.4
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.11.1
IPython          : 7.13.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.4.1
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

**Comment by mroeschke** [comment]
Looks to be fixed on master:

```
In [1]: import datetime as dt
   ...: import pandas as pd
   ...:
   ...: df = pd.DataFrame({'foo': [dt.datetime(2000, 1, 1)]})
   ...: df.at[0, 'foo'] = dt.datetime(2000, 1, 2, tzinfo=dt.timezone.utc)
   ...: print(df.at[0, 'foo'])  # Prints 2000-01-02 00:00:00
   ...: assert df.at[0, 'foo'].tzinfo is not None
2000-01-02 00:00:00+00:00

In [2]: pd.__version__
Out[2]: '1.1.0.dev0+1253.g556e1c26a'
```

I find similar but not explicit tests for this case (DataFrame assignment with `datetime.datetime` object). Could use a regression test

**Comment by EdAbati** [comment]
Hi @mroeschke , can I pick this up (if it is still available)?

I could not find tests on the method `.at` but I found this: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/frame/indexing/test_iat.py

I could add the case in this issue in a test, similarly to what has been done on `.iat` in the past.
Is it what you were referring to or did you have something else in mind?

**Comment by mroeschke** [comment]
Sure @EdAbati. That location for the new test sounds good.

**Comment by simonjayhawkins** [comment]
> Looks to be fixed on master:

fixed in #31182

4edcc5541ff3f6470f5e3c083cb83136119e6f0c is the first new commit
commit 4edcc5541ff3f6470f5e3c083cb83136119e6f0c
Author: jbrockmendel <jbrockmendel@gmail.com>
Date:   Mon Jan 27 22:11:28 2020 -0800

    CLN: Make Series._values match Index._values (#31182)

**Comment by simonjayhawkins** [comment]
xref #22740
